addons/hr_homeworking/models/hr_homeworking.py

```diff
@@ -0,0 +1,293 @@
+DAYS = ['monday_location_id', 'tuesday_location_id', 'wednesday_location_id', 'thursday_location_id', 'friday_location_id', 'saturday_location_id', 'sunday_location_id']
+
+class HrHomeworking(models.Model):
+    _name = "hr.homeworking"
+
+    def _default_employee(self):
+        if 'default_employee' in self.env.context:
+            return self.env.context.get('default_employee')
+        return self.env['hr.employee'].search([('user_id.id','=', self.env.user.id)])
+        
+    # work location
+    work_location_id = fields.Many2one('hr.work.location', required=True)
+    work_location_name = fields.Char(related='work_location_id.name')
+    work_location_type = fields.Selection(related="work_location_id.location_type")
+    #relative to event
+    employee_id = fields.Many2one('hr.employee', default=_default_employee, required=True)
+    weekday = fields.Integer()
+    weekly = fields.Boolean(default=False)
+    date = fields.Date(help="Start date for a default worklocation", string="Start Date")
+    #relative to event supression
+    removed = fields.Boolean(default=False, help="True if a specific date on a default record is removed")
+    parent_default_homeworking = fields.Many2one("hr.homeworking", help="removed records from default record")
+    child_removed_ids = fields.One2many("hr.homeworking",'parent_default_homeworking')
+    #relative to previous default worklocation
+    current = fields.Boolean(default=True, help="useful only if it's a default worklocation or old default")
+    end = fields.Date(help="useful only if it's an old default worklocation to save the past")
+    _sql_constraints = [
+        ('uniq_name', 'unique(employee_id, weekday, current)', 'Only one current default work location per day per employee. can have some previous default work location'),
+        ('uniq_truc', 'unique(employee_id, weekly, date, current)', 'Only one default work location and one exceptional work location per day per employee.'),
+    ]
+
+    def _save_previous_default_worklocation(self):
+        if self.date > fields.Date().today():
+            # self.child_removed_ids.unlink()
+            # self.unlink()
+            return True
+        else:
+            self.env.cr.execute(
+                'UPDATE hr_homeworking SET current=NULL WHERE id=%s',
+                (self.id,)
+            )
+            self.end = fields.Date().today() - + timedelta(days=1)
+            self.child_removed_ids.filtered(lambda a : a.date >= fields.Date.today()).unlink()
+            return False
+
+
+    def _next_weekday_date_after_today(self, date):
+        return fields.Date().today() + timedelta(days= ((date.weekday() - fields.Date.today().weekday() +7)%7))
+
+    def _removed_worklocation_to_active_worklocation(self, work_location_id):
+        self.removed = False
+        self.parent_default_homeworking = False
+        self.work_location_id = work_location_id
+
+    def action_register_new_day_work_location(self):
+        print("coucou")
```

Comment:
That's because you defined the view of models.Model as a transient, which requires a method call to be saved. 

Just define your view as a classic form view (no footer)

Reply:
Yes but if I don't put the footer , i don't know how change the name of the button "create" to "set location"